pytorch_lightning/plugins/training_type/dp.py

```diff
@@ -101,10 +102,16 @@ def predict_step(self, *args, **kwargs):
         return self.model(*args, **kwargs)
 
     def training_step_end(self, output):
-        return self.reduce(output)
+        if not is_overridden("training_step_end", self.lightning_module):
```

Comment:
@SeanNaren this check for DP should take care of the previous rule we had: only auto reduce if user did not do it themself.